qiskit/transpiler/passes/optimization/commutative_inverse_cancellation.py

```diff
@@ -23,19 +23,20 @@
 class CommutativeInverseCancellation(TransformationPass):
     """Cancel pairs of inverse gates exploiting commutation relations."""
 
-    def __init__(self, max_qubits: int = 4, upto_phase_optimization: bool = False):
+    def __init__(self, matrix_based: bool = False, max_qubits: int = 4):
         """Initialize CommutativeInverseCancellation pass.
 
         Args:
-            max_qubits: limits the number of qubits in matrix-based commutativity and inverse
-                checks.
-            upto_phase_optimization: if True, also cancels out pairs of gates that are
-                inverse up to a phase, while keeping track of the phase difference in the
-                global phase of the circuit. However, this inverse check is now matrix-based
-                and becomes more expensive.
+            matrix_based: if True, uses matrix representations to check whether two
+                operations are inverse of each other. This makes the checks more powerful,
+                and in addition allows canceling pairs of operations that are inverse up to a
+                phase, while updating the global phase of the circuit accordingly.
+                Generally this leads to more reductions at the expense of increased runtime.
+            max_qubits: limits the number of qubits in matrix-based commutativity and
+                inverse checks.
```

Comment:
Docstring nit-picks 🙂 
```suggestion
            matrix_based: If ``True``, uses matrix representations to check whether two
                operations are inverse of each other. This makes the checks more powerful,
                and, in addition, allows canceling pairs of operations that are inverse up to a
                phase, while updating the global phase of the circuit accordingly.
                Generally this leads to more reductions at the expense of increased runtime.
            max_qubits: Limits the number of qubits in matrix-based commutativity and
                inverse checks.
```

Reply:
Done in [f955fe7](https://github.com/Qiskit/qiskit/pull/11248/commits/f955fe75f5518a63ddd808059b51115866fb0ecf).